### PR TITLE
jasmin: init at 2.4

### PR DESCRIPTION
**Pull request by fgaz**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/compilers/jasmin/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  installPhase =
+  ''
+    install -Dm644 jasmin.jar $out/share/java/jasmin.jar
+    mkdir -p $out/bin
+    makeWrapper ${jre}/bin/java $out/bin/jasmin \
+      --add-flags "-jar $out/share/java/jasmin.jar"
+  '';
+
+  meta = with stdenv.lib; {
+    description = "an assembler for the Java Virtual Machine";
```
Comment (JohnAZoidberg) [inline pkgs/development/compilers/jasmin/default.nix]:
```suggestion
    description = "An assembler for the Java Virtual Machine";
```

File: pkgs/development/compilers/jasmin/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+, fetchurl
+, unzip
+, jdk
+, ant
+, makeWrapper
+, jre
+}:
+
+let
+  d2u = stdenv.lib.replaceChars ["."] ["_"];
```
Comment (JohnAZoidberg) [inline pkgs/development/compilers/jasmin/default.nix]: This isn't used anywhere, is it?
```suggestion
```
(Remove the let/in, too, please.)

**Review by JohnAZoidberg — CHANGES_REQUESTED** [review]
Does it work, did you try to assemble a binary?

Is this maybe a fork worth looking into, since the original hasn't been updated for a decade? https://github.com/Sable/jasmin

**Comment by JohnAZoidberg** [comment]
I tried it. The hello world example [here](http://jasmin.sourceforge.net/about.html) can be assembled and run like:

```
> ./result/bin/jasmin HelloWorld.jasmin
Generated: HelloWorld.class
> java -cp . HelloWorld
Hello World!
```

I think it'd be cool, to have a test that tests this, using the `test` attribute: https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

File: pkgs/development/compilers/jasmin/default.nix (hunk 1 above)
Comment (fgaz) [reply pkgs/development/compilers/jasmin/default.nix]: Fixed.

File: pkgs/development/compilers/jasmin/default.nix (hunk 2 above)
Comment (fgaz) [reply pkgs/development/compilers/jasmin/default.nix]: Oops, that was a leftover from a previous version. Fixed.

**Comment by fgaz** [comment]
About the fork: I'm not sure. It says "a specialized version of jasmin which is used as a back end for the Soot framework" and the original project does not link to it, so I'm a bit hesitant. Maybe we can merge this and decide later whether to upgrade?

About test: cool, I didn't know about that. Is it new? I'll add it shortly.

**Comment by fgaz** [comment]
Also most distros do not package that fork: https://repology.org/project/jasmin/versions

**Comment by fgaz** [comment]
Added the test

**Review by JohnAZoidberg — APPROVED** [review]
Awesome!
Yeah, the fork looks a bit dodgy.

**Pull request merged**